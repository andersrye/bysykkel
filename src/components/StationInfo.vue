<template>
  <div>
    <div
      class="station-title"
      data-test="title"
    >
      {{ station?.name }}
    </div>
    <div 
      class="station-info"
      data-test="address"
    >
      {{ station?.address }}
    </div>
    <div class="station-info">
      <span class="station-count">
        <FontAwesomeIcon icon="bicycle" />
        <span
          class="station-count-number"
          data-test="bikes-available"
        >{{ station?.num_bikes_available }}</span>
      </span>
      <span class="station-count">
        <FontAwesomeIcon icon="lock-open" />
        <span
          class="station-count-number"
          data-test="docks-available"
        >{{ station?.num_docks_available }}</span>
      </span>
    </div>
  </div>
</template>

<script setup>
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'

defineProps({
  station: { type: Object, default: () => ({}) }
})
</script>

<style scoped>
.station-title {
  font-size: 16px;
}
.station-info {
  font-size: 12px;
  color: #6f6f6f;
}
.station-count {
  display: inline-block;
  margin: .2em .9em .2em .1em;
}
.station-count-number {
  display: inline-block;
  width: 1.3em;
  margin-left: .4em;
}
</style>
