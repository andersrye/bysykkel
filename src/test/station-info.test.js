import { mount } from '@vue/test-utils'
import StationInfo from '../components/StationInfo.vue'
import { expect, test } from 'vitest'
import station from './test-data/station-combined.json'

test('mount component', async () => {
    const wrapper = mount(StationInfo, {
        props: {station},
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.get('[data-test=title]').text()).toContain('TÃ¸yen skole')
    expect(wrapper.get('[data-test=address]').text()).toContain('Ringgata 2A')
    expect(wrapper.get('[data-test=bikes-available]').text()).toContain('2')
    expect(wrapper.get('[data-test=docks-available]').text()).toContain('13')

})