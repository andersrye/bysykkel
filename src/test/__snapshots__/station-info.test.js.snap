// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div data-v-45f68d02=\\"\\">
  <div class=\\"station-title\\" data-test=\\"title\\" data-v-45f68d02=\\"\\">TÃ¸yen skole</div>
  <div class=\\"station-info\\" data-test=\\"address\\" data-v-45f68d02=\\"\\">Ringgata 2A</div>
  <div class=\\"station-info\\" data-v-45f68d02=\\"\\"><span class=\\"station-count\\" data-v-45f68d02=\\"\\"><!----><span class=\\"station-count-number\\" data-test=\\"bikes-available\\" data-v-45f68d02=\\"\\">2</span></span><span class=\\"station-count\\" data-v-45f68d02=\\"\\"><!----><span class=\\"station-count-number\\" data-test=\\"docks-available\\" data-v-45f68d02=\\"\\">13</span></span></div>
</div>"
`;
